lib/floe/workflow/runner/kubernetes.rb

```diff
@@ -102,6 +102,33 @@ def cleanup(runner_context)
           delete_secret(secret) if secret
         end
 
+        def wait(timeout: nil, events: %i[create update delete])
+          watcher = kubeclient.watch_pods(:namespace => namespace)
+
+          timeout_thread = Thread.new { sleep(timeout); watcher.finish } if timeout.to_i > 0
+
+          watcher.each do |notice|
+            event = kube_notice_type_to_event(notice.type)
+            next unless events.include?(event)
+
+            pod = notice.object
+            container_ref   = pod.metadata.name
+            container_state = pod.to_h.deep_stringify_keys["status"]
```

Comment:
Why the deep_stringify_keys here? Seems like an expensive operation just to get the status, where instead you could do

```suggestion
            pod_h = pod.to_h
            container_state = pod_h.fetch("status", pod_h[:status])
```

Or perhaps just hit the pod object direct with something like `pod.status`

Reply:
We have issues with symbol keys and serializing to jsonb and back, that said this could be `pod.to_h[:status].deep_stringify_keys` instead